src/lib/logical_query_plan/lqp_translator.cpp

```diff
@@ -212,10 +212,12 @@ std::shared_ptr<AbstractOperator> LQPTranslator::_translate_predicate_node_to_in
   // TableScan(s).
   auto index_scan = std::make_shared<IndexScan>(input_operator, SegmentIndexType::GroupKey, column_ids,
                                                 predicate->predicate_condition, right_values, right_values2);
+  index_scan->set_included_chunk_ids(indexed_chunks);
 
-  const auto table_scan = _translate_predicate_node_to_table_scan(node, input_operator);
+  // If all chunks of a table are indexed, we don't need anything else except the IndexScan
```

Comment:
Negative. At translation time, we don't necessarily have a transaction id. Chunks might still be added before we execute the scan and those should not be excluded. In fact, we should add a test making sure that results from new chunks are included.

Reply:
Right. In this case, the changes to the LQPTranslator and the added test are obsolete.